app/models/ext_management_system.rb

```diff
@@ -422,6 +422,47 @@ def enable!
     update!(:enabled => true)
   end
 
+  def self.destroy_queue(ids)
+    ids = ids.to_miq_a
+    _log.info("Queuing destroy of #{name} with the following ids: #{ids.inspect}")
+    ids.each do |id|
+      MiqQueue.put(
+        :class_name  => name,
+        :instance_id => id,
+        :method_name => "orchestrate_destroy",
+      )
+    end
+  end
+
+  # override destroy_queue from AsyncDeleteMixin
+  def destroy_queue(deliver_on = nil)
+    _log.info("Queuing destroy of #{self.class.name} with id: #{id}")
+    MiqQueue.put(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "orchestrate_destroy",
+      :deliver_on  => deliver_on,
+    )
+  end
+
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    if self.destroy == false
+      _log.info("Cant #{self.class.name} with id: #{id}, workers still in progress. Requeuing destroy...")
+      raise MiqException::MiqQueueRetryLater.new(:deliver_on => 15.seconds.from_now)
```

Comment:
we rarely use `MiqQueueRetryLater` and are going away from that since it is queue system specific. Can we just queue another destroy?